FirebaseVertexAI/Sources/FirebaseInfo.swift

```diff
@@ -0,0 +1,41 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+import FirebaseAppCheckInterop
+import FirebaseAuthInterop
+import FirebaseCore
+
+/// Firebase data used by VertexAI
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+struct FirebaseInfo {
+  let appCheck: AppCheckInterop?
+  let auth: AuthInterop?
+  let projectID: String
+  let apiKey: String
+  let googleAppID: String?
```

Comment:
Could we have a let `let isDataCollectionDefaultEnabled: Bool` and then make `googleAppID` non-optional?

Then we could change this below
```
if let googleAppID = firebaseInfo.googleAppID {
  urlRequest.setValue(googleAppID, forHTTPHeaderField: "X-Firebase-AppId")
}
```
to
```
if firebaseInfo.isDataCollectionDefaultEnabled {
  urlRequest.setValue(firebaseInfo.googleAppID, forHTTPHeaderField: "X-Firebase-AppId")
}
```

If this is just temporary then I think it's fine but my main concern is that I might not remember when `googleAppID` should be set or not.

Reply:
Good point. Fixed.